### ISSUE TITLE
Crash on startup [Wayland]

### ISSUE THREAD

**Issue by my4ng** [flathub/com.discordapp.Discord#413]
The small update window appears for a moment and then discord crashes. It works on KDE (wayland) but not niri (wayland).

```
$ flatpak run --socket=wayland com.discordapp.Discord --enable-features=UseOzonePlatform --ozone-platform=wayland

Disabling updates already done
Discord 0.0.58
Starting app.
Starting updater.
2024-07-04T01:52:21.390Z [Modules] Modules initializing
2024-07-04T01:52:21.390Z [Modules] Distribution: remote
2024-07-04T01:52:21.390Z [Modules] Host updates: disabled
2024-07-04T01:52:21.390Z [Modules] Module updates: enabled
2024-07-04T01:52:21.390Z [Modules] Module install path: /home/my4ng/.var/app/com.discordapp.Discord/config/discord/0.0.58/modules
2024-07-04T01:52:21.390Z [Modules] Module installed file path: /home/my4ng/.var/app/com.discordapp.Discord/config/discord/0.0.58/modules/installed.json
2024-07-04T01:52:21.390Z [Modules] Module download path: /home/my4ng/.var/app/com.discordapp.Discord/config/discord/0.0.58/modules/pending
splashScreen.initSplash(false)
CDM component API found
blackbox: 2024-07-04T01:52:21.488Z 0

----------------------------------------------
blackbox: 2024-07-04T01:52:21.500Z 1 Discord starting: {"releaseChannel":"stable","version":"0.0.58"}, modulepath: /home/my4ng/.var/app/com.discordapp.Discord/config/discord/0.0.58/modules
blackbox: 2024-07-04T01:52:21.502Z 2 ✅ webContents.created web1 "index.html"
blackbox: 2024-07-04T01:52:21.502Z 3 ✅ window.created win1 "discord"
splashScreen: SPLASH_SCREEN_READY
2024-07-04T01:52:21.551Z [Modules] No updates to install
splashScreen: no-pending-updates
2024-07-04T01:52:21.551Z [Modules] Host is up to date.
2024-07-04T01:52:21.551Z [Modules] Checking for module updates at https://discord.com/api/modules/stable/versions.json
splashScreen: checking-for-updates
splashScreen.updateSplashState checking-for-updates checking-for-updates {}
splashScreen.webContentsSend: SPLASH_UPDATE_STATE SPLASH_UPDATE_STATE [ { status: 'checking-for-updates' } ]
2024-07-04T01:52:21.557Z [Modules] No module updates available.
splashScreen: update-check-finished true 0 false
CDM completed with status (old updater): [{"id":"oimompecagnajdejgnnjijobebaeigek","status":"new","title":"Widevine Content Decryption Module","version":"4.10.2710.0"}]
splashScreen.launchMainWindow: false
Optional module ./ElectronTestRpc was not included.
splashScreen.updateSplashState launching launching {}
splashScreen.webContentsSend: SPLASH_UPDATE_STATE SPLASH_UPDATE_STATE [ { status: 'launching' } ]
blackbox: 2024-07-04T01:52:21.625Z 4 ✅ webContents.created web2 ""
blackbox: 2024-07-04T01:52:21.628Z 5 ✅ window.created win2 "Discord"
blackbox: 2024-07-04T01:52:21.632Z 6 ✅ webContents.did-finish-load web1
[WebContents] crashed (reason: crashed, exitCode: 11)... reloading
Optional module ./ElectronTestRpc was not included.
blackbox: 2024-07-04T01:52:22.145Z 7 ❌ render-process-gone { reason: 'crashed', exitCode: 11 }
blackbox: 2024-07-04T01:52:22.154Z 8 window.closed win2
blackbox: 2024-07-04T01:52:22.155Z 9 ✅ webContents.created web3 ""
blackbox: 2024-07-04T01:52:22.156Z 10 ✅ window.created win3 "Discord"
blackbox: 2024-07-04T01:52:22.156Z 11 webContents.destroyed web2
[WebContents] double crashed (reason: crashed, exitCode: 11)... RIP =(
notificationScreen.webContentsSend: win is invalid undefined.
blackbox: 2024-07-04T01:52:22.555Z 12 ❌ render-process-gone { reason: 'crashed', exitCode: 11 }
blackbox: 2024-07-04T01:52:22.561Z 13 before-quit
2024/07/04 11:52:22 socat[3] W exiting on signal 15
```

**Comment by YaLTeR** [comment]
Discord wants X11 for some reason; on KDE it probably accesses KDE's Xwayland socket (it's allowed in the Flatpak manifest), but on niri there's none.

**Comment by my4ng** [comment]
> Discord wants X11 for some reason; on KDE it probably accesses KDE's Xwayland socket (it's allowed in the Flatpak manifest), but on niri there's none.

Indeed with xwayland-satellite running in the background and setting the DISPLAY env var, it opens without any issue. This is bizarre since under KDE it states that it is running as a native wayland window using the debug console.

**Comment by ghishadow** [comment]
what error you getting in ```journalctl```? , check using ```journalctl -r```, for me it was crashing on discord_utils macos notifications, it worked after removing discord cofig

**Comment by my4ng** [comment]
> what error you getting in `journalctl`? , check using `journalctl -r`, for me it was crashing on discord_utils macos notifications, it worked after removing discord cofig

I did not get any get any error in there. I think because it is an application issue rather than a flatpak one.

**Comment by eggsnpie** [comment]
Having this same issue on Gnome 46 with Wayland. It's frustrating to say the least.

**Comment by itspngu** [comment]
Same here with `sway` on Gentoo. Happy it's the flatpak/app that's broken for once rather than my system :^)

**Comment by Operativ** [comment]
Discord is still crashing on Fedora 40 using KDE, Wayland.

If this app uses some parts of Chromium, then it would also be wise to check for the recently added default flags that make Chromium inspect CPU, GPU and Memory (which SELinux does not allow doing, and might crash the app).

**Comment by FlailoftheLord** [comment]
Same issue.
Plasma 6.1.4 on Wayland 
Linux 6.10.5-arch1-1
Solution: Reset flatseal permissions to default and make sure it's only pointing to X11 and not wayland or fallback x11

**Comment by andrewd18** [comment]
A potential fix for this on Fedora systems has been pushed into kernel 6.10.6, now available in stable. See: https://bugzilla.redhat.com/show_bug.cgi?id=2254434

**Comment by eggsnpie** [comment]
The issue actually stopped for me. I installed the Discord app in the past week and it works fine now. I don't know what changed.

**Comment by guihkx** [comment]
Thank you for confirming. What distro are you on, by the way?

I have an old GPU that doesn't support Wayland, so I've set up an Arch Linux virtual machine instead to use it, but I couldn't reproduce the crash:

![Screenshot_20241004_131304](https://github.com/user-attachments/assets/d8330344-d9f1-44dc-bf9b-9f390315a743)


It'd be great if you or someone else could check if the non-Flatpak Discord package also crashes.

**Comment by codesxt** [comment]
It is also crashing for me in Debian 10 with the .deb version.

**Comment by guihkx** [comment]
@codesxt Thank you for that info, but I think root the cause of your issue might be different than the one originally reported (even though the error is the same).

I've set up a Debian 10 VM (which is unsupported, btw, and you should upgrade), then I installed Discord from the official deb package, and I got the same crash as you did. All of this on XFCE4 + X11, so not Wayland-related.

Funnily enough, running Flatpak'd Discord on this old Debian version seems to work great, which makes me suspect that this crash is graphics-related. I'd like to try upgrading mesa to see if the crash would go away, but accomplishing such task seems virtually impossible on such old version of Debian.

And even though I don't think "we" (as a Flatpak package) can do anything to prevent this crash (except, well, **not** enabling the Wayland socket), I'd still like to know if this crash occurs on newer, supported distros.

**Comment by lambdaknight** [comment]
It happens on AlmaLinux 8.10 (Cerulean Leopard)

**Comment by guihkx** [comment]
@lambdaknight Please post the output of `flatpak run com.discordapp.Discord`

**Comment by barzoj** [comment]
Same issue on openSUSE Tumbleweed, installed discord rpm from official repo. 

`journalctl -r`:

```
Oct 16 09:07:55 home drkonqi-coredump-launcher[1990945]: Nothing handled the dump :O
Oct 16 09:07:55 home drkonqi-coredump-launcher[1990945]: Unable to find file for pid 1990716 expected at "kcrash-metadata/Discord.2adcca9b6b334acdbb31ce57ab616df0.1990716.ini"
Oct 16 09:07:55 home systemd[2679]: Started Launch DrKonqi for a systemd-coredump crash (PID 1990924/UID 0).
Oct 16 09:07:55 home systemd-coredump[1990923]: [🡕] Process 1990716 (Discord) of user 1000 dumped core.
                                                
                                                Module /home/iddqd/.config/discord/0.0.70/modules/discord_zstd/discord_zstd.node without build-id.
                                                Module /home/iddqd/.config/discord/0.0.70/modules/discord_zstd/discord_zstd.node
                                                Module /home/iddqd/.config/discord/0.0.70/modules/discord_utils/node_modules/macos-notification-state/build/Release/notificationstate.node wi>
                                                Stack trace of thread 1990920:
                                                #0  0x00007f25f590ef31 av_dict_set (libffmpeg.so + 0x10ef31)
                                                ELF object binary architecture: AMD x86-64
Oct 16 09:07:55 home (sd-parse-elf)[1990925]: Could not parse number of program headers from core file: invalid `Elf' handle
```
discord log:

```
Discord 0.0.70
Starting app.
Starting updater.
2024-10-16T07:14:22.591Z [Modules] Modules initializing
2024-10-16T07:14:22.591Z [Modules] Distribution: remote
2024-10-16T07:14:22.591Z [Modules] Host updates: disabled
2024-10-16T07:14:22.591Z [Modules] Module updates: enabled
2024-10-16T07:14:22.591Z [Modules] Module install path: /home/iddqd/.config/discord/0.0.70/modules
2024-10-16T07:14:22.591Z [Modules] Module installed file path: /home/iddqd/.config/discord/0.0.70/modules/installed.json
2024-10-16T07:14:22.591Z [Modules] Module download path: /home/iddqd/.config/discord/0.0.70/modules/pending
splashScreen.initSplash(false)
CDM component API found
blackbox: 2024-10-16T07:14:22.645Z 0 

----------------------------------------------
CDM completed with status: cdm-ready-success
blackbox: 2024-10-16T07:14:22.650Z 1 Discord starting: {"releaseChannel":"stable","version":"0.0.70"}, modulepath: /home/iddqd/.config/discord/0.0.70/modules
blackbox: 2024-10-16T07:14:22.653Z 2 ✅ webContents.created web1 "index.html"
blackbox: 2024-10-16T07:14:22.653Z 3 ✅ window.created win1 "discord"
splashScreen: SPLASH_SCREEN_READY
2024-10-16T07:14:22.706Z [Modules] No updates to install
splashScreen: no-pending-updates
2024-10-16T07:14:22.706Z [Modules] Host is up to date.
2024-10-16T07:14:22.706Z [Modules] Checking for module updates at https://discord.com/api/modules/stable/versions.json
splashScreen: checking-for-updates
splashScreen.updateSplashState checking-for-updates checking-for-updates {}
splashScreen.webContentsSend: SPLASH_UPDATE_STATE SPLASH_UPDATE_STATE [ { status: 'checking-for-updates' } ]
2024-10-16T07:14:22.799Z [Modules] No module updates available.
splashScreen: update-check-finished true 0 false
splashScreen.launchMainWindow: false
Optional module ./ElectronTestRpc was not included.
splashScreen.updateSplashState launching launching {}
splashScreen.webContentsSend: SPLASH_UPDATE_STATE SPLASH_UPDATE_STATE [ { status: 'launching' } ]
blackbox: 2024-10-16T07:14:22.828Z 4 ✅ webContents.created web2 ""
blackbox: 2024-10-16T07:14:22.830Z 5 ✅ window.created win2 "Discord"
blackbox: 2024-10-16T07:14:23.243Z 6 ❌ render-process-gone { reason: 'crashed', exitCode: 139 }
legacyModuleUpdater: installed-module discord_krisp 1 1 true
legacyModuleUpdater: installed-module discord_game_utils 1 1 true
mainScreen.UpdaterEvents: UPDATER_HISTORY_QUERY_AND_TRUNCATE
mainScreen.on(did-finish-load) false false
blackbox: 2024-10-16T07:14:23.724Z 7 ✅ webContents.did-finish-load web2
splashScreen.pageReady
blackbox: 2024-10-16T07:14:23.824Z 8 window.close win1
blackbox: 2024-10-16T07:14:23.827Z 9 webContents.destroyed web1
blackbox: 2024-10-16T07:14:23.827Z 10 window.closed win1
mainScreen.UpdaterEvents: UPDATER_HISTORY_QUERY_AND_TRUNCATE
mainScreen.UpdaterEvents: UPDATER_HISTORY_QUERY_AND_TRUNCATE
mainScreen.UpdaterEvents: CHECK_FOR_UPDATES UPDATE_NOT_AVAILABLE
2024-10-16T07:14:23.884Z [Modules] Host is up to date.
2024-10-16T07:14:23.884Z [Modules] Checking for module updates at https://discord.com/api/modules/stable/versions.json
2024-10-16T07:14:23.886Z [Modules] No module updates available.
legacyModuleUpdater: update-check-finished
legacyModuleUpdater: installed-module discord_rpc 1 1 true
mainScreen.UpdaterEvents: UPDATER_HISTORY_QUERY_AND_TRUNCATE
[WebContents] crashed (reason: crashed, exitCode: 139)... reloading
Optional module ./ElectronTestRpc was not included.
blackbox: 2024-10-16T07:14:25.320Z 11 ❌ render-process-gone { reason: 'crashed', exitCode: 139 }
blackbox: 2024-10-16T07:14:25.324Z 12 window.closed win2
blackbox: 2024-10-16T07:14:25.328Z 13 ✅ webContents.created web3 ""
blackbox: 2024-10-16T07:14:25.329Z 14 ✅ window.created win3 "Discord"
blackbox: 2024-10-16T07:14:25.330Z 15 webContents.destroyed web2
legacyModuleUpdater: installed-module discord_krisp 1 1 true
legacyModuleUpdater: installed-module discord_game_utils 1 1 true
mainScreen.UpdaterEvents: UPDATER_HISTORY_QUERY_AND_TRUNCATE
mainScreen.UpdaterEvents: UPDATER_HISTORY_QUERY_AND_TRUNCATE
mainScreen.on(did-finish-load) false true
blackbox: 2024-10-16T07:14:26.109Z 16 ✅ webContents.did-finish-load web3
splashScreen.pageReady
splashScreen.destroySplash: splashWindow is null or destroyed.
mainScreen.UpdaterEvents: UPDATER_HISTORY_QUERY_AND_TRUNCATE
mainScreen.UpdaterEvents: CHECK_FOR_UPDATES UPDATE_NOT_AVAILABLE
2024-10-16T07:14:26.208Z [Modules] Host is up to date.
2024-10-16T07:14:26.208Z [Modules] Checking for module updates at https://discord.com/api/modules/stable/versions.json
2024-10-16T07:14:26.210Z [Modules] No module updates available.
legacyModuleUpdater: update-check-finished
legacyModuleUpdater: installed-module discord_rpc 1 1 true
mainScreen.UpdaterEvents: UPDATER_HISTORY_QUERY_AND_TRUNCATE
[WebContents] double crashed (reason: crashed, exitCode: 139)... RIP =(
notificationScreen.webContentsSend: win is invalid undefined.
blackbox: 2024-10-16T07:14:27.646Z 17 ❌ render-process-gone { reason: 'crashed', exitCode: 139 }
blackbox: 2024-10-16T07:14:27.649Z 18 before-quit
```

**Comment by Oughie** [comment]
I can start an Xwayland-Session manually using `Xwayland $DISPLAY &` to temporarily fix this issue. Seems `Xwayland` wasn't running for me at all, so I was also unable to run a few other applications.

**Comment by Oughie** [comment]
I started an Xwayland-Session manually using `Xwayland $DISPLAY &` to temporarily fix this issue. Seems `Xwayland` wasn't running for me at all, so I was also unable to run a few other applications as well, including Discord.

**Comment by Oughie** [comment]
I started an Xwayland-Session manually using `Xwayland $DISPLAY &` to temporarily fix this issue. Apparently `Xwayland` wasn't running for me at all, so I was unable to run Discord as well as some other applications.

**Comment by 20after4** [comment]
I don't know if this will help anyone else but for me the issue was resolved by using Flatseal to make the following changes:


### Socket:
#### Enable:
```
socket=x11
socket=wayland
```
#### Disable:
```
socket=fallback-x11
```

### Device:
#### Enable:
```
device=dri
```

**Comment by DeedleFake** [comment]
@20after4, isn't that just using X11, though? `fallback-x11` is a pseudo-socket that has Flatpak check if Wayland is available before starting the app, then give it _only_ Wayland access if it is and fall back to X11 if it isn't. Turning that off means that Discord will have both Wayland and X11 access, so it'll just use X11.

**Comment by 20after4** [comment]
@DeedleFake for whatever reason that is what worked for me. I am running [xwayland-satellite](https://github.com/Supreeeme/xwayland-satellite), however, Discord doesn't seem to be using an x11 window.

*There is probably a better way to test which windows are using XWayland, however, I think this method is accurate:*
  
I ran `xwininfo` in a terminal and then tried to click on the Discord window to select it but xwininfo doesn't react, it's still waiting for me to click. This method does work with Steam's window, the only one I know for sure is using XWayland.

**Comment by DeedleFake** [comment]
My favorite way to check for an Xwayland window is to use xeyes. If the eyes can track the cursor, it's over an Xwayland window.

**Comment by DeedleFake** [comment]
I just was able to test it myself. You are correct, it prevents the crash _and_ creates a Wayland window. Interesting. It must be connecting to X11 for something even when using a Wayland window. Maybe it has to do with push-to-talk.

**Comment by DeedleFake** [comment]
After some more testing I can confirm that that is, at least in part, what is happening. I have little tool I made, https://github.com/DeedleFake/ptt-fix, that gets around the push-to-talk issues on Wayland by directly listening to the evdev input devices and then sending button presses to the root X window so that Xwayland windows can detect them, even with Wayland running. Even with Discord using a Wayland window, push-to-talk with another window focused works with the tool running and does _not_ work without it running, meaning that Wayland Discord is still using X11 to listen for global push-to-talk button presses.
